tools/oas-apigee-mock/lib/commands/generateApi/generateProxyEndPoint.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var builder = require('xmlbuilder')
+var fs = require('fs')
+var path = require('path')
+var serviceUtils = require('../../util/service.js')
+var assignMessage = require('../../policy_templates/assign-message/assign-message.js')
+
+module.exports = function generateProxyEndPoint(apiProxy, options, api, cb) {
+  var useCors
+  var destination = options.destination || path.join(__dirname, '../../../api_bundles')
+  if (destination.substr(-1) === '/') {
+    destination = destination.substr(0, destination.length - 1)
+  }
+
+  var rootDirectory = destination + '/' + apiProxy + '/apiproxy'
+  var root = builder.create('ProxyEndpoint')
+  root.att('name', 'default')
+  root.ele('Description', {}, api.info.title)
+  var preFlow = root.ele('PreFlow', { name: 'PreFlow' })
+
+  // Add steps to preflow.
+  var raiseFaultName
+  var requestPipe = preFlow.ele('Request')
+  var responsePipe = preFlow.ele('Response')
+  var services = serviceUtils.servicesToArray(api)
+
+  var flows = root.ele('Flows', {})
+
+  var allowedVerbs = ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS', 'HEAD', 'TRACE', 'CONNECT', 'PATCH']
```

Comment:
This should be configurable via options or left out. Right now all possible verbs are listed here anyway.